### PR TITLE
chore: Add action to warn about potentially risky PR changes

### PR DESCRIPTION
**Pull request by kahest**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [x] Internal

## :scroll: Description

Implement a mechanism that automatically adds an "alert" comment to PRs if they include changes to files we think have higher potential to break things than others.

## :bulb: Motivation and Context

same as https://github.com/getsentry/sentry-java/pull/3726 and https://github.com/getsentry/sentry-dart/pull/2315

## :green_heart: How did you test it?
✅ by adding the file-filter.yml introduced in this PR to the list of potentially risky changes and ensuring that an alert is shown
Screenshot 2024-09-30 at 17 25 14

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] ~I added tests to verify changes~
- [ ] ~No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled~
- [ ] ~All tests passing~
- [ ] ~No breaking changes~

## :crystal_ball: Next steps

#skip-changelog

### CODE REVIEW

File: .github/file-filters.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# This is used by the action https://github.com/dorny/paths-filter
+
+high_risk_code: &high_risk_code
+  - '.github/file-filters.yml'
```
Comment (kahest) [inline .github/file-filters.yml:4]: - this is for testing and will be removed before merging
- need to align on set of potentially higher-risk files before merging as well
- use full repo paths, can check [getsentry/sentry-cocoa@main/.github/file-filters.yml](https://github.com/getsentry/sentry-cocoa/blob/main/.github/file-filters.yml?rgh-link-date=2024-09-30T06%3A52%3A52Z) for inspiration

Comment (krystofwoldrich) [reply .github/file-filters.yml:4]: I think a good candidates are

- https://github.com/getsentry/sentry-react-native/blob/933804ec2ad2153b9bd42d91bff2bb69d7f2a672/packages/core/src/js/NativeRNSentry.ts
  - Represents the native module interface, when changes usually we see new issue about failed build due to outdated pods for example 
- https://github.com/getsentry/sentry-react-native/blob/ddcabffc8bd015392f28deb4f685ed358889c77e/packages/core/scripts/sentry-xcode.sh and https://github.com/getsentry/sentry-react-native/blob/ddcabffc8bd015392f28deb4f685ed358889c77e/packages/core/sentry.gradle
  - automatic upload scripts currently covered only by e2e build tests
  - missing unit tests at the moment

Comment (lucas-zimerman) [inline .github/file-filters.yml:4]: I also agree that the Native references on TypeScript are good candidates for it, what do you think @krystofwoldrich?
```suggestion
  - '.github/file-filters.yml'
  - 'src/js/vendor/react-native/index.ts'
  - 'src/js/touchevents.tsx'
  - 'src/js/utils/rnlibraries.ts'
  - 'src/js/utils/rnlibrariesinterface.ts'
  - 'src/js/NativeRNSentry.ts'
```

**Review by krystofwoldrich — APPROVED** [review]
LGTM 🚀

**Review by lucas-zimerman — APPROVED** [review]
LGTM!

**Pull request merged**